src/logic/activities.js

```diff
@@ -299,6 +299,38 @@ export const roundsMissingScorecards = wcif =>
     .filter(round => groupActivitiesAssigned(wcif, round.id))
     .filter(round => parseActivityCode(round.id).eventId !== '333fm');
 
+export const competitorsRegisteredForAnEventWithoutGroups = wcif => {
+  let persons = [];
+  wcif.persons.forEach(person => {
+    if (!person.registration || person.registration?.status !== 'accepted')
+      return;
+    person.registration.eventIds.forEach(eventId => {
+      const hasCompetitorAssignment = person.assignments.some(
+        a =>
+          a.assignmentCode === 'competitor' &&
+          activityById(wcif, a.activityId).activityCode.startsWith(
+            `${eventId}-r1`
+          )
+      );
+      if (!hasCompetitorAssignment) {
+        if (!persons.some(p => p.person.wcaUserId === person.wcaUserId)) {
```

Comment:
Having `persons` as a list means we need to go through it repetitively with `.some` and `.find`. We can make this more efficient, by having `persons` as a map, so we can do this:

```js
if (!persons[person.wcaUserId]) {
  persons[person.wcaUserId] = { person, eventIds: [] };
}
```

and then

```js
persons[person.wcaUserId].eventIds.push(eventId)
```

This makes the operations O(1) instead of O(n).

At the end we can return `Object.values(persons)` to get a list.